fastestimator/estimator.py

```diff
@@ -339,34 +362,48 @@ def _run_epoch(self, eager: bool) -> None:
                                ds_id=self.system.ds_id,
                                steps_per_epoch=self.system.steps_per_epoch,
                                output_keys=trace_input_keys - network_output_keys | network_input_keys) as loader:
+
+                self.system.eval_log_steps = []
```

Comment:
if we're adding a field to the system object, that should be done within the system object itself rather than by the estimator. Creating a new member variable here and then referencing it elsewhere will be confusing for future development and IDE linting won't work well. Also, if this variable is really required, then it should also probably be given a slightly different name, for example progress_log_steps or something to make it more clear what it's for